### ISSUE TITLE
Add plugin support

### ISSUE THREAD

**Issue by Proxtx** [spckio/spck-issues#46]
It would be great if you would add an easy plugin support. For example:
`spck.on("edit", (file, edit)=>{
if(file.line[edit.line].word[edit.word] == "nl"){
file.line.context += "\n";
})
`

**Comment by Proxtx** [comment]
[Spck - Plugin.pdf](https://github.com/spckio/spck-issues/files/3903040/Spck.-.Plugin.pdf)

An idea for plugins in spck editor

**Comment by EduApps-CDG** [comment]
I like this idea

**Comment by adr420** [comment]
Amazing idea ,

__Benefits__  ==>

1. It would be too easy for devs also to add features via plugins , they can also have some premium plugins.
2. Also it will be easy for users too , to have a choice between what features they want and what they don't.

All i can say there are just benefits of this feature , i don't think if there's any downside of it ...but at the moment the app is not that stable to have features like this one.

**Comment by 1-AlenToma** [comment]
agreed, I was using acode an switched to this as it have better intalicense. but now I am missing the color generator and other plugins acode offer. 

this is a must for this app to get and stay on top. 

otherwise good app
